csrc/apis/c/rotation_detector.cpp

```diff
@@ -0,0 +1,150 @@
+  // clang-format on
+  return v;
+}
+
+template <class ModelType>
+int mmdeploy_rotation_detector_create_impl(ModelType&& m, const char* device_name, int device_id,
+                                           mm_handle_t* handle) {
+  try {
+    auto value = config_template();
+    value["pipeline"]["tasks"][0]["params"]["model"] = std::forward<ModelType>(m);
+
+    auto pose_estimator = std::make_unique<Handle>(device_name, device_id, std::move(value));
+
+    *handle = pose_estimator.release();
+    return MM_SUCCESS;
+
+  } catch (const std::exception& e) {
+    MMDEPLOY_ERROR("exception caught: {}", e.what());
+  } catch (...) {
+    MMDEPLOY_ERROR("unknown exception caught");
+  }
+  return MM_E_FAIL;
+}
+
+}  // namespace
+
+int mmdeploy_rotation_detector_create(mm_model_t model, const char* device_name, int device_id,
+                                      mm_handle_t* handle) {
+  return mmdeploy_rotation_detector_create_impl(*static_cast<Model*>(model), device_name, device_id,
+                                                handle);
+}
+
+int mmdeploy_rotation_detector_create_by_path(const char* model_path, const char* device_name,
+                                              int device_id, mm_handle_t* handle) {
+  return mmdeploy_rotation_detector_create_impl(model_path, device_name, device_id, handle);
+}
+
+int mmdeploy_rotation_detector_apply(mm_handle_t handle, const mm_mat_t* mats, int mat_count,
+                                     mm_rotate_detect_t** results, int** result_count) {
+  if (handle == nullptr || mats == nullptr || mat_count == 0 || results == nullptr) {
+    return MM_E_INVALID_ARG;
+  }
+
+  try {
+    auto detector = static_cast<Handle*>(handle);
+
+    Value input{Value::kArray};
+    for (int i = 0; i < mat_count; ++i) {
+      mmdeploy::Mat _mat{mats[i].height,         mats[i].width, PixelFormat(mats[i].format),
+                         DataType(mats[i].type), mats[i].data,  Device{"cpu"}};
+      input.front().push_back({{"ori_img", _mat}});
+    }
+
+    auto output = detector->Run(std::move(input)).value().front();
+    auto detector_outputs = from_value<vector<mmrotate::RotationDetectorOutput>>(output);
+
+    vector<int> _result_count;
+    _result_count.reserve(mat_count);
+    for (const auto& det_output : detector_outputs) {
+      _result_count.push_back((int)det_output.detections.size());
+    }
+
+    auto total = std::accumulate(_result_count.begin(), _result_count.end(), 0);
+
+    std::unique_ptr<int[]> result_count_data(new int[_result_count.size()]{});
+    auto result_count_ptr = result_count_data.get();
+    std::copy(_result_count.begin(), _result_count.end(), result_count_data.get());
+
+    auto deleter = [&](mm_rotate_detect_t* p) {
+      mmdeploy_rotation_detector_release_result(p, result_count_ptr);
+    };
+    std::unique_ptr<mm_rotate_detect_t[], decltype(deleter)> result_data(
+        new mm_rotate_detect_t[total]{}, deleter);
+    // ownership transferred to result_data
```

Comment:
Incorrect comment since ownership is taken by `result_count_ptr`. 
B.T.W, why not use `auto result_count_ptr = result_count_data.release()` so that L114-115 can be removed.